node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1023,6 +950,11 @@ pub(crate) async fn run<Context>(
 				FromOrchestra::Signal(Conclude) => return Ok(()),
 			},
 			(relay_parent, peer_id) = state.active_collation_fetches.select_next_some() => {
+				for authority_id in state.peer_ids.get(&peer_id).into_iter().flatten() {
+					// This peer is no longer interested in this relay parent.
+					state.validator_groups_buf.reset_validator_bit(relay_parent, authority_id);
```

Comment:
This is probably fine because we don't immediately take effect on this, but the above future also fires when the fetch was not yet concluded, but also when it exceeded its [unshared bandwith duration](https://github.com/paritytech/polkadot/blob/70bffc7365777089f6f84af524cb0c2938cbeddb/node/network/collator-protocol/src/collator_side/mod.rs#L643). Which means the fetch might be still in progress, would be bad to disconnect.

Reply:
This timeout is supposed to be the upper limit isn't it? If the peer is slow there's no point in keeping the connection alive.

The case is a corner one anyway, what should happen is that we distribute a new collation right after the timeout is hit.

Reply:
The timeout hits after 150ms ... A peer might actually be quite fast and still hit the timeout. "Timeout" is in fact a bad naming here, because it isn't really a timeout - it is just, ok let's give one guy a head start so it can have unshared bandwidth for a while, so it can start validating asap. At the same time we don't want to delay propagating to other peers too long, so we start uploads one after the other in 150ms intervals. 

The general idea being: It is better if 2 validators get a full collation, than 5 validators getting 2/5 of a collation. Especially because if one validator has it all, it will start propagating to others as well, which is good for two reasons:

1. Multiplied bandwidth: This validator's bandwidth is added to the collators bandwidth for further propagation.
2. Potentially reduced latency: Let's assume that the validator that received the collation is a bit closer to one of the other validators than our collator. In that case it might have a smaller round trip time and can forward the collation faster, even if bandwidth is not actually the issue, but rather latency.

TL;DR: The timeout hitting, does neither mean that the validator is done, nor that it necessarily is bad. It is rather a normal condition that is planned to happen on everyday network conditions.

Reply:
I can back-port changes from async backing branch where we actually catch a timeout:
https://github.com/paritytech/polkadot/blob/0b0c70a3f7b8839a06080d141c680e4104403eb1/node/network/collator-protocol/src/collator_side/mod.rs#L1212-L1221

And only disconnect if the collation was actually fetched